### PR TITLE
eth/handler: enforce lookup limits in handler

### PR DESCRIPTION
**Pull request by holiman**
adds some more limits to the eth handler

### CODE REVIEW

File: eth/handler.go [hunk 1]
```diff
@@ -410,8 +410,12 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 			bytes   common.StorageSize
 			headers []*types.Header
 			unknown bool
+			lookups int
 		)
-		for !unknown && len(headers) < int(query.Amount) && bytes < softResponseLimit && len(headers) < downloader.MaxHeaderFetch {
+		for !unknown && len(headers) < int(query.Amount) && bytes < softResponseLimit &&
+			len(headers) < downloader.MaxHeaderFetch &&
+			lookups < 2*downloader.MaxHeaderFetch {
```
Comment (fjl) [inline eth/handler.go:417]: Whoa, so many conditions.

File: eth/handler.go [hunk 2]
```diff
@@ -410,8 +410,12 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
...
+			lookups++
```
Comment (rjl493456442) [inline eth/handler.go:418]: Is the `lookups` always equal to `len(headers)`?

File: eth/handler.go [hunk 3]
```diff
@@ -612,6 +622,7 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 				data = append(data, entry)
 				bytes += len(entry)
 			}
+			lookups++
```
Comment (rjl493456442) [inline eth/handler.go:625]: Why bumping the `lookups` twice?

File: eth/handler.go (hunk 2 above)
Comment (holiman) [reply eth/handler.go:418]: Yes, I suppose it will be in this particular clause

File: eth/handler.go (hunk 3 above)
Comment (holiman) [reply eth/handler.go:625]: Error on my part!

**Comment by holiman** [comment]
Closing in favour of https://github.com/karalabe/go-ethereum/pull/36

**Pull request closed without merging**